File: frontend/src/app/corporation/corporation.ts
import { Character } from '../character/character';

export class Corporation {

  name: string;

  corporationID: number;

  allianceID: number;

  CEOID: number;

  ticker: string;

  members: Map<number, Character> = new Map();

  expiresOn: Date;

  getId(): number {
    return this.corporationID;
  }

}

export class Wallets {

  corporationID: number;
  divisions: Array<Wallet>;

}

export class Wallet {
  balance: number;
  division: number;
}

File: model/corporation.go
package model

import (
	"fmt"
	"time"
)

type Corporation struct {
	expireDate      *time.Time
	CorporationID   int32            `json:"corporationID"`
	CorporationName string           `json:"Name"`
	AllianceID      int32            `json:"allianceID"`
	CEOID           int32            `json:"CEOID"`
	Ticker          string           `json:"ticker"`
	Members         map[string]int32 `json:"members"`
}

func (c *Corporation) ID() int32 {
	return c.CorporationID
}

func (c *Corporation) ExpiresOn() *time.Time {
	return c.expireDate
}

func (c *Corporation) SetExpire(t *time.Time) {
	c.expireDate = t
}

func (c *Corporation) HashKey() string {
	return fmt.Sprintf("corporation:%d", c.ID())
}
